src/ORM/Association.php

```diff
@@ -1279,6 +1289,22 @@ protected function _extractFinder($finderData)
     }
 
     /**
+     * Gets the table class name.
+     *
+     * @param string $alias The alias name you want to get.
+     * @param array $options Table options array.
+     * @return string
+     */
+    protected function _getClassName($alias, array $options = [])
+    {
+        if (empty($options['className'])) {
+            $options['className'] = Inflector::camelize($alias);
```

Comment:
@lorenzo I'm afraid we can't rely on locator's "duplicate check" here. I've tried this.
`TableLocatror` checks if already registered table has exactly the same build options as a requested one.

For example:

```php
TableRegistry::get('Bar.Foo'); 
TableRegistry::get('Foo', ['className' => 'Bar.Foo']);
TableRegistry::get('Foo', ['className' => 'Bar\Model\Table\FooTable']);
```

Theoretically same requests but `RuntimeException` would be thrown.